src/Analysis/Ast/Impl/Analyzer/Symbols/SymbolCollector.cs

```diff
@@ -104,12 +105,16 @@ internal sealed class SymbolCollector : PythonWalker {
         }
 
         private void AddFunction(FunctionDefinition fd, IPythonType declaringType) {
-            if (!(_eval.LookupNameInScopes(fd.Name, LookupOptions.Local) is PythonFunctionType existing)) {
+            var existing = _eval.LookupNameInScopes(fd.Name, LookupOptions.Local) as PythonFunctionType;
+            if (existing is null) {
```

Comment:
It is better to preserve type check instead of introducing `null` check:
```cs
if (_eval.LookupNameInScopes(fd.Name, LookupOptions.Local) is PythonFunctionType existing) {
    ReportRedefinedFunction(fd, existing);
} else {
    existing = new PythonFunctionType(fd, declaringType, _eval.GetLocationOfName(fd));
    // The variable is transient (non-user declared) hence it does not have location.
    // Function type is tracking locations for references and renaming.
    _eval.DeclareVariable(fd.Name, existing, VariableSource.Declaration);
}
```

Reply:
Yes this is much nicer syntax, thanks!